src/programs/sum_all_eer_files/sum_all_eer_files.cpp

```diff
@@ -0,0 +1,215 @@
+        output_gain_filename = my_input->GetFilenameFromUser("Output gain file name", "Filename of output gain image", "gain_image.mrc", false);
+    }
+
+    max_threads = my_input->GetIntFromUser("Max number of threads to use", "maximum number of threads to use for processing.", "1", 1);
+    eer_super_res_factor = my_input-> GetIntFromUser("EER super resolution factor", "super resolution factor for pixels.", "1", 1);
+    eer_frames_per_image = my_input->GetIntFromUser("EER frames per image", "frames per image in eer file type", "0", 0);
+
+    delete my_input;
+
+    my_current_job.Reset(7);
+    my_current_job.ManualSetArguments("tbttiii", output_filename.c_str( ), make_dark_and_gain, output_dark_filename.c_str( ), output_gain_filename.c_str( ), max_threads, eer_super_res_factor, eer_frames_per_image);
+}
+
+// override the do calculation method which will be what is actually run..
+
+bool SumAllEer::DoCalculation( ) {
+
+    long frame_counter;
+    long file_counter;
+    long pixel_counter;
+    long total_summed = 0;
+
+    int file_x_size;
+    int file_y_size;
+
+    std::string output_filename      = my_current_job.arguments[0].ReturnStringArgument( );
+    bool        make_dark_and_gain   = my_current_job.arguments[1].ReturnBoolArgument( );
+    std::string output_dark_filename = my_current_job.arguments[2].ReturnStringArgument( );
+    std::string output_gain_filename = my_current_job.arguments[3].ReturnStringArgument( );
+    int         max_threads          = my_current_job.arguments[4].ReturnIntegerArgument( );
+    int			eer_super_res_factor = my_current_job.arguments[5].ReturnIntegerArgument( );
+    int 		eer_frames_per_image = my_current_job.arguments[6].ReturnIntegerArgument( );
+
+    wxArrayString all_files;
+    wxDir::GetAllFiles(".", &all_files, "*.eer", wxDIR_FILES);
+    all_files.Sort( );
+
+    EerFile* current_input_file;
+
+    Image output_sum_image;
+    Image output_dark_image;
+    Image output_gain_image;
+
+    // global and local for threading...
+
+    double* global_sum_image;
+    double* global_sum_squares_image;
+
+    double* sum_image;
+    double* sum_squares_image;
+
+    // find all the Eer files in the current directory..
+
+    wxPrintf("\nThere are %li eer files in this directory.\n", all_files.GetCount( ));
+
+    current_input_file = new EerFile();
+    current_input_file->OpenFile(all_files.Item(0).ToStdString( ), false, false, false, eer_super_res_factor, eer_frames_per_image);
+
+    file_x_size = current_input_file->ReturnXSize( );
+    file_y_size = current_input_file->ReturnYSize( );
+
+    wxPrintf("\nFirst file is %s\nIt is %ix%i sized - all images had better be this size!\n\n", all_files.Item(0), current_input_file->ReturnXSize( ), current_input_file->ReturnYSize( ));
+
+    delete current_input_file;
+
+    output_sum_image.Allocate(file_x_size, file_y_size, 1);
+    output_sum_image.SetToConstant(0.0);
```

Comment:
The compiler gets it I'm sure - but using the float literal is standard practice

```cpp
output_sum_image.SetToConstant(0.f)
// or
output_sum_image.SetToConstant(0.0f)
```